storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java

```diff
@@ -56,6 +56,7 @@ public class LeaderEpochFileCache {
      * @param topicPartition the associated topic partition
      * @param checkpoint     the checkpoint file
      */
+    @SuppressWarnings("this-escape")
     public LeaderEpochFileCache(TopicPartition topicPartition, LeaderEpochCheckpoint checkpoint) {
```

Comment:
This is probably due to calling `this.assign()` which is a private method. The code works because it is the last line in the ctor but it's a legitimate warning. Should we create a ticket for fixing this correctly (and removing the warning)?

Reply:
The problem is that we pass `this` to `forEach`. Not sure there's a lot of value in fixing this.